src/lib/alloc.c

```diff
@@ -268,7 +269,7 @@ static void *alloc_cont_blocks(struct mm_heap *heap, int level,
 		map->first_free += count;
 
 	/* update each block */
-	for (current = start; current < count; current++) {
+	for (i = 0, current = start; i < count; i++, current++) {
```

Comment:
Why would we update the state of blocks that has not been affected by this continuous allocation? I assume your intention is to lock everything before that newly created continuous one, but I doubt it. There is a possibility that there is some space for allocation for smaller blocks which these change will block, right?